### ISSUE TITLE
introduce specialized "binary" versions of loss functions to support class weights

### ISSUE THREAD

**Issue by Kiyoshika** [Kiyoshika/Generalized-Model-Framework#32]
right now, most loss functions (besides cross entropy) are kind of generic to support both classification and regression tasks.

however, with the recent work on computing class weights, maybe it's a good idea to "specialize" these loss functions and create binary versions to implement class weight support.

for example, `gmf_loss_squared_binary` could be something like
```c
return one_weight * target * (1.0f - estimate) * (1.0f - estimate) 
           + zero_weight * (1.0f - target) * (0.0f - estimate) * (0.0f - estimate);
```
and so on for the other loss functions.
